vm/actor/src/builtin/market/mod.rs

```diff
@@ -329,15 +310,18 @@ impl Actor {
                 .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
 
             for id in &params.deal_ids {
-                let mut deal: DealState = states
+                let deal = states
                     .get(*id)
-                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?
-                    .ok_or_else(|| {
-                        ActorError::new(
-                            ExitCode::ErrIllegalState,
-                            "Failed to retrieve the DealState".to_owned(),
-                        )
-                    })?;
+                    .map_err(|e| ActorError::new(ExitCode::ErrIllegalState, e.into()))?;
+
+                if deal.is_some() {
```

Comment:
You can also do

deal.ok_or_else(|_|ActorError::new(ExitCode::ErrIllegalArgument,format!("given deal already included in another sector: {}", id)))?